mono/mini/mini.c

```diff
@@ -4272,3 +4272,20 @@ mini_class_is_system_array (MonoClass *klass)
 {
 	return m_class_get_parent (klass) == mono_defaults.array_class;
 }
+
+int mono_target_pagesize (void)
+{
+	static int pagesize = -1;
+
+	if (G_LIKELY (pagesize != -1))
+		return pagesize;
+
+	mono_memory_barrier ();
+#ifdef MONO_CROSS_COMPILE
+	// use a safe lower bound
+	pagesize = 4 * 1024;
```

Comment:
We could probably go with higher values for some targets.

Reply:
I tried 4k with roslyn, even that doesn't get beyond one 4k page for implicit null checking. let's be conservative here, increasing this value doesn't buy us anything.

Reply:
Makes sense, let's keep it that way.